Microsoft.Toolkit.Forms.UI.XamlHost/IDesktopWindowXamlSourceNative2.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using WUX = Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Win32.UI.XamlHost
+{
+    /// <summary>
+    /// Enables access to native methods on DesktopWindowXamlSourceNative version 2 for RS5
+    /// </summary>
+    /// <remarks>
+    /// Includes the method used to set the window handle of the <see cref="WUX.Hosting.DesktopWindowXamlSource" /> instance.
+    /// </remarks>
+    [ComImport]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    [Guid("e3dcd8c7-3057-4692-99c3-7b7720afda31")]
+    internal partial interface IDesktopWindowXamlSourceNative2
```

Comment:
WinForms and WPF require different version of the message struct.

---
In reply to: [260995968](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/85#discussion_r260995968) [](ancestors = 260995968)

Reply:
99% of the files are the same. Maybe a partial class with the common code and two spefici for it would be better. Or in this case, since it is literally just one param type that is different, a simple conditional compilation #if statement might be better. What do you think?

Reply:
A partial class for a COM interface is risky as we need to guarantee the exact same code layout, PreTranslateMessage MUST be in the 3rd slot of the V-Table, I dont think that partial classes can guarantee the v table layout.


---
In reply to: [261032440](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/85#discussion_r261032440) [](ancestors = 261032440)